web/srv/api_handlers.go

```diff
@@ -438,3 +438,22 @@ func (h *handler) handleAPIResourceDefinition(w http.ResponseWriter, req *http.R
 	w.Header().Set("Content-Type", "text/yaml")
 	w.Write(resourceDefinition)
 }
+
+func (h *handler) handleAPIGateways(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	requestParams := util.GatewayRequestParams{
+		TimeWindow:        req.FormValue("window"),
+		GatewayNamespace:  req.FormValue("gatewayNamespace"),
+		RemoteClusterName: req.FormValue("remoteClusterName"),
+	}
+	gatewayRequest, err := util.BuildGatewayRequest(requestParams)
```

Comment:
I don't think we really need a brand new struct and helper method for building the api request.  The only properties we need are the time window and (optionally) the gateway namespace so we should be able to construct the request inline here easily enough.

Reply:
to make it consistent with the code with other handler's. I added this, but one more reason for adding this is that in future if we try to get a particular gateway from a namespace

Reply:
I'm a big fan consistency but I think in this case it's more valuable to simplify and to reduce boilerplate.  If we need more functionality in the future we can always add it at that time.

Reply:
sure @adleong will do the changes

Reply:
> sure @adleong will do the changes, but I am wondering the `timeWindow` validation should be done at the helper method. So, I don't want to pollute the `handler` by putting all the logic in a single place.
> 
> And if I add the `timeWindow` validation at the handler level, then I need to introduce one more constant `defaultTimeWindow`.And, I don't want this @adleong

Reply:
I don't think the validation is complex enough to require a separate function.  This code is more readable to me to have the time window validation and defaulting in-line here in the handler.

Reply:
@adleong, can I remove both the `BuildGatewayRequest` for now ?

Reply:
yes, that's what I'm suggesting.

Reply:
@adleong removed it